recipes/usockets/all/conanfile.py

```diff
@@ -0,0 +1,120 @@
+import os
+
+from conans import ConanFile, tools, MSBuild
+from conans.errors import ConanInvalidConfiguration
+
+
+class UsocketsConan(ConanFile):
+    name = "usockets"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Miniscule cross-platform eventing, networking & crypto for async applications"
+    license = "Apache-2.0"
+    homepage = "https://github.com/uNetworking/uSockets"
+    topics = ("conan", "socket", "network", "web")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"fPIC": [True, False],
+               "with_ssl": [False, "openssl", "wolfssl"],
+               "with_libuv": [True, False]}
+    default_options = {"fPIC": True, "with_ssl": False, "with_libuv": True}
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("Windows build is not supported by upstream")
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.libcxx
+        if self.options.with_ssl == "wolfssl":
+            self.options["wolfssl"].opensslextra = True
```

Comment:
Overriding options is not okay, because that can cause confusion if a consumer sets `wolfssl:opensslextra=False` in its profile.
Better is to add the following test in *build** (not configure)
```
if self.options.with_ssl == "wolfssl":
    if not wolfssl.opensslextra:
        raise ConanInvalidConfiguration("wolfssl needs opensslextra option enabled for usockets")
```

Reply:
Good point, thanks. I will fix it soon

Reply:
By the way, why in `build` but not in `configure`? Usually checking options goes to `configure`.

Reply:
The options of dependencies are not final at that point. Conan sets them to `None`.
Try for yourself by printing its value in the configure method.

Reply:
Got it, thanks